contracts/feature-tests/composability/promises-features/src/call_promise_direct.rs

```diff
@@ -40,13 +41,14 @@ pub trait CallPromisesDirectModule {
 
         let gas_limit = (self.blockchain().get_gas_left() - extra_gas_for_callback) * 9 / 10;
 
-        self.send()
-            .contract_call::<()>(to, endpoint_name)
-            .with_multi_token_transfer(token_payments_vec)
-            .with_gas_limit(gas_limit)
+        self.tx()
+            .to(&to)
+            .raw_call(endpoint_name)
+            .egld_or_multi_esdt(EgldOrMultiEsdtPayment::MultiEsdt(token_payments_vec))
```

Comment:
.payment(token_payments_vec)

We only use EgldOrMultiEsdtPayment if the choice is at runtime